colcon_mixin/mixin/__init__.py

```diff
@@ -13,19 +14,36 @@
 
 mixins_by_verb = None
 
+"""Environment variable to read additional mixins from"""
+COLCON_MIXIN_PATH = EnvironmentVariable(
+    'COLCON_MIXIN_PATH',
+    'Provide additional directories to look for mixin files. '
+    'Separate individual directories with colons.')
+
 
 def get_mixin_path():
     """
-    Get the path where mixin is stored.
+    Get the path where mixins are stored in the COLCON_HOME configuration.
 
     :rtype: Path
     """
     return get_config_path() / 'mixin'
 
 
+def get_additional_mixin_paths():
+    """
+    Get the paths where additional mixins may be found.
+
+    :rtype: list
+    """
+    # Read additional paths from the environment variable, splitting on :.
+    env_var = os.environ.get(COLCON_MIXIN_PATH.name, '')
+    return [Path(x) for x in env_var.split(':') if x]
```

Comment:
This won't work correctly on Windows. You should try `os.pathsep` and update the comment above.
```suggestion
    return [Path(x) for x in env_var.split(os.pathsep) if x]
```

Reply:
Thanks for the review @cottsay , I didn't know about `os.pathsep`, thanks for pointing it out, I've updated the PR as recommended. :+1: